storm-client/src/jvm/org/apache/storm/serialization/KryoTupleDeserializer.java

```diff
@@ -39,7 +38,7 @@ public KryoTupleDeserializer(final Map<String, Object> conf, final GeneralTopolo
         _kryoInput = new Input(1);
     }        
 
-    public Tuple deserialize(byte[] ser) {
+    public TupleImpl deserialize(byte[] ser) {
```

Comment:
added @Override. 
As we were expecting (and downcasting) its return value anyway in some places, .. used covariance instead. Intent was to avoid unnecessary casting  in the critical path. (which requires rttti).